WpfUtility/Services/ObservableObject.cs

```diff
@@ -74,5 +109,38 @@ protected bool SetField<T>(ref T field, T value, [CallerMemberName] string prope
 
             return true;
         }
+
+        /// <summary>
+        /// Gets all properties of the deriving class
+        /// </summary>
+        /// <param name="accessType">The <see cref="PropertyAccessType"/> of the properties (optional, default = Read | Write)</param>
+        /// <param name="onlyPublic"><see langword="true"/> to get only public properties, otherwise <see langword="false"/> (optional, default = <see langword="true"/></param>
+        /// <returns></returns>
+        protected List<string> GetProperties(PropertyAccessType accessType = PropertyAccessType.Read | PropertyAccessType.Write, bool onlyPublic = true)
+        {
+            if (!GetType().IsClass)
+                return new List<string>();
+
+            List<PropertyInfo> properties;
+            switch ((int)accessType)
+            {
+                case 1:
+                    properties = GetType().GetProperties().Where(w => w.CanRead).ToList();
+                    break;
+                case 2:
+                    properties = GetType().GetProperties().Where(w => w.CanWrite).ToList();
+                    break;
+                case 3:
+                    properties = GetType().GetProperties().Where(w => w.CanRead && w.CanWrite).ToList();
```

Comment:
Sollte das nen "Und" oder ein "Oder" sein?

Reply:
Is this case even needed? Because you only have "PropertyAccessType.Read | PropertyAccessType.Write" and if the property can be read and/or written case 1 or 2 already took place

Reply:
Nevermind, this is fine the way it is